### ISSUE TITLE
stdcm: map shakes when adjusting time

### ISSUE THREAD

**Issue by Castavo** [OpenRailAssociation/osrd#10144]
### What happened?

When clicking on the time adjusting buttons of the form (among other inputs), the map "shakes" I guess it's reloading.

https://github.com/user-attachments/assets/d656b357-44d2-4e48-aa33-9f9fa9b9b719

### What did you expect to happen?

I expect the map to either stay stable or to not reload at all when editing the time component

### How can we reproduce it (as minimally and precisely as possible)?

1. Start a STDCM request
2. Fill in the origin
3. Fill in a waypoint / the destination (in order to have two pins placed on the map)
4. Now adjust the time component as shown on video
5. Witness

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Firefox

### OSRD version (top right corner `Account` button > `Informations`)

6580cfe

**Comment by theocrsb** [comment]
I can't reproduce the bug. Is it still there?

**Comment by Castavo** [comment]
This might be obscured by #10240

**Comment by sim51** [comment]
The map is recenter automatically when a user changes : 
- traction engine 
- time
- date 
- tolerance
- A CH code
- a CI
-  ...

Question : does the map should be automatically recenter on the "path" when the form's modifications don't impact it ?

**Comment by sim51** [comment]
I respond to my question : yes we need to recenter it (except for the time, date & tolerance).
When an input is changed, we call the API to find a path between the origin/destination, and the found path can be completely different than the previous one, and so the BBOX can changed. 

I can reproduce  it with : 

* engine : 26000US - BB 26000
* Origin: valence 
* Destination: Miramas
* via : orange

Bug explanation : 
- when we change a field, a call is made to find a path (see `useStaticPathfinding`).  
- during this process the pathfinding is set to `undefined` before to be set it with the API return
- For each new instance of the pathfinding (even undefined), the map is redraw and the BBOX computed. And if the path is `undefined`, the BBOX is computed with the `markersInfo`

So when we change the time field, we have one map render with a bbox for the marker, and a second later with pathfinding bbox.  The blink effect comes from those 2 renders which can be different.

**Comment by thibautsailly** [comment]
This blinking doesn't scream "quality". Ideally, there should be a map state of "computing new path", and a transition between the two displays rather than a sudden move when the new path becomes available.
As a first step we can prevent it from happening when time related data is edited, and improve upon it later.

**Comment by SharglutDev** [comment]
I don't think we should recompute the pathfinding when we change an input not related to it (tolerance, date, time)
